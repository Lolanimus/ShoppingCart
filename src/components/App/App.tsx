import React, { useState } from "react";
import styles from "./App.module.scss";

const App = () => {
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  };

  return (
    <div className={styles.app}>
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </div>
  );
};

export default App;
