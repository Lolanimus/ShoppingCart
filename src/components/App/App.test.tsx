import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  });
});