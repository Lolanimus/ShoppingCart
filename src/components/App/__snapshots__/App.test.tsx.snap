// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders magnificent monkeys 1`] = `
<div>
  <div
    class="_app_bbbd1d"
  >
    <button
      type="button"
    >
      Click Me
    </button>
    <h1>
      Magnificent Monkeys
    </h1>
  </div>
</div>
`;
